### ISSUE TITLE
Feature: Wait-Time for next update

### ISSUE THREAD

**Issue by Thomas--F** [SSilence/selfoss#750]
Hi Folks,

every time I call the feed-update, every feed is fetched. But I have a lot of feeds, that updates only once a day or once a week (e.g. some online-cartoons) while others updates very frequently (e.g. news-feeds). It would be great, if I can set a wait-option at a source, that says:
Skip this feed during update until a given time period is expired.

There is already a "lastupdate"-column in sources. So we need a new update-period-column.
I can code the changes, but only if this feature is wanted. Changing the database shouldn't be done without a good reason.

**Comment by niol** [comment]
Better, the update period could be computed from past entries: no configuration, no database change.

**Comment by Thomas--F** [comment]
That wouldn't run well. You have to enable this per feed, so you must save the config somewhere.
Otherwise the update will go crazy, if a feed updates very randomly.

I implement this feature on my RasPi and it works fine so far. I added a column to the database where you can save the minimum wait time in seconds, until this feed is fetched again.
I have some feeds on "0 seconds", so the update every time, the cronjob runs (in my case every 10 minitues). Others run on 3600 seconds (once an hour) or even 86400 seconds (once a day).

The workload on the RaspberryPie has dropped a lot after the changes.

**Comment by niol** [comment]
I don't see your point about the update going _crazy_ and I doubt many people will take the time to manualy configure this per feed.

**Comment by martinsauter** [comment]
I think it should really be configurable per feed as I have feeds for example that update only once a day or only once every 2 or 3 days. Nevertheless I still want Selfoss to look for an update during every run because I want to find out that there is something new **IMMEDIATELY** and not only much later just because there is only a new post every couple of days. 

From my point of view the frequency with which to check for news in a feed and the frequency in which a new post appears in a feed have **nothing** to do with each other in many cases.

**Comment by jtojnar** [comment]
If an average time interval between *last n items* is used it might be unreliable when items don’t follow homogeneous distribution (i.e. articles are published in the evening).

It could record a release times and compute the probability distribution of updates but that would probably be an overkill.

@Thomas--F In your implementation, is this a property of RSS spout or is it possible to configure it for any spout?

**Comment by Thomas--F** [comment]
@niol The default ist "0 seconds waittime", so if you don't configure anything, everything is running the old way

In my implementation, it is a property of every feed (an additional Integer-column "waitperiod" in the table "sources", just like lastentry and lastupdate). It stores the wait-time in seconds, so I just had to add a simple if-statement to the update-function in ContentLoader.php
`if (($last + $wait) < time()){`
`       $this->fetch($source);`
`}`

**Comment by bbeardsley** [comment]
+1 many of my feeds aren't updated that often so would save bandwidth

**Comment by thomwiggers** [comment]
RSS feeds also have `ttl` tags and tags like `skipHours` and `skipDays` that could be used to more efficiently poll feeds http://www.rssboard.org/rss-specification#optionalChannelElements.

**Comment by jtojnar** [comment]
@thomwiggers it would be nice to have the whole specification implemented. Although I would be surprised if any feed ever used `ttl`.

**Comment by Thomas--F** [comment]
I found another usecase of my implementation. 
I can nearly disable a feed by setting the wait time to 999.999.999 Seconds (about 31 years)

**Comment by Thomas--F** [comment]
I tested my implemention of this feature now for some weeks and it looks stable enough.
There's only one thing, that stops me from forking and make a pull request:
There has to be an additional integer-column in the sources-table. I called it "waitperiod".
How do I implement the changes in the database? Is there some kind of "update"-function for this?

**Comment by PhrozenByte** [comment]
@Thomas--F: See the classes in `daos/*/Database.php`, particularly the `__construct()` methods.

**Comment by Thomas--F** [comment]
So, it's done! Feel free to test it.

@PhrozenByte: Thanks for the hint!
